addon/src/motions/adjust-css.ts

```diff
@@ -12,43 +13,59 @@ import { rAF } from '../-private/concurrency-helpers';
   @param {Object} options
   @return {Motion}
 */
-export default function adjustCSS(propertyName, sprite, opts) {
+export default function adjustCSS(
+  propertyName: keyof CopiedCSS,
+  sprite: Sprite,
+  opts: Partial<AdjustCSSOptions> = {},
+) {
   return new AdjustCSS(propertyName, sprite, opts).run();
 }
 
-adjustCSS.property = function (propertyName) {
+adjustCSS.property = function (propertyName: keyof CopiedCSS) {
   return this.bind(null, propertyName);
 };
 
-export class AdjustCSS extends Motion {
-  constructor(propertyName, sprite, opts) {
+interface AdjustCSSOptions extends BaseOptions {
+  easing: (time: number) => number;
+}
+
+export class AdjustCSS extends Motion<AdjustCSSOptions> {
+  prior: AdjustCSS | null | undefined = null;
+  tween: TweenLike | null = null;
+
+  constructor(
+    readonly propertyName: keyof CopiedCSS,
+    sprite: Sprite,
+    opts: Partial<AdjustCSSOptions> = {},
+  ) {
     super(sprite, opts);
-    this.propertyName = propertyName;
-    this.prior = null;
-    this.tween = null;
   }
 
-  interrupted(motions) {
+  interrupted(motions: Motion[]) {
     this.prior = motions.find(
       (m) => m instanceof AdjustCSS && m.propertyName === this.propertyName,
-    );
+      // SAFETY: We just checked that it's an instance of AdjustCSS
+    ) as AdjustCSS | undefined;
   }
 
   *animate() {
     let { value: finalValue, unit } = this._splitUnit(
-      this.sprite.finalComputedStyle[this.propertyName],
+      this.sprite.finalComputedStyle![this.propertyName],
     );
     if (this.prior) {
+      let prior: AdjustCSS = this.prior;
+      prior.assertHasTween();
```

Comment:
Annoyingly, TS requires that we assign it to a variable _and_ declare the type to use the assertion.